source/read/retrieve.txt

```diff
@@ -0,0 +1,238 @@
+      "crunchbase_url" : "http://www.crunchbase.com/company/linkedin", "homepage_url" : "http://linkedin.com",
+      ...
+   
+.. tip:: Sort Order
+
+   The ``find_one()`` method returns the first document in 
+   :manual:`natural order </reference/glossary/#std-term-natural-order>`
+   on disk if no sort criteria is specified.
+
+.. TODO:
+ To learn more about sorting, see the :ref:`sort guide <cpp-sort>`.
+
+.. _cpp-retrieve-find-multiple:
+
+Find Multiple Documents
+~~~~~~~~~~~~~~~~~~~~~~~
+
+To find multiple documents in a collection, pass a query filter to the ``find()``
+method that specifies the criteria of the documents you want to retrieve. 
+
+The following example uses the ``find()`` method to find all documents in which
+the ``founded_year`` field has the value ``1970``:
+
+.. literalinclude:: /includes/read/retrieve.cpp
+    :language: cpp
+    :dedent:
+    :start-after: start-find-many
+    :end-before: end-find-many
+
+The ``find()`` method returns an instance of ``mongocxx::cursor``, which you can
+iterate over to see the matching documents. A cursor is a mechanism that allows an
+application to iterate over database results while holding only a subset of them in
+memory at a given time. Cursors are useful when your ``find()`` method returns a large
+amount of documents.
+
+You can iterate over the documents in a cursor by using a ``for-in`` loop, as shown in
+the following example:
+
+.. io-code-block::
+
+   .. input:: /includes/read/retrieve.cpp
+      :start-after: start-cursor
+      :end-before: end-cursor
+      :language: cpp
+      :dedent:
+
+   .. output::
+
+      { "_id" : { "$oid" : "52cdef7d4bab8bd675298be4" }, "name" : "Mitsubishi Motors",
+      "permalink" : "mitsubishi-motors", "crunchbase_url" : "http://www.crunchbase.com/company/mitsubishi-motors",
+      ...
+
+      { "_id" : { "$oid" : "52cdef7e4bab8bd67529b996" }, "name" : "Western Digital",
+      "permalink" : "western-digital", "crunchbase_url" : "http://www.crunchbase.com/company/western-digital",
+      ...
+
+      { "_id" : { "$oid" : "52cdef7e4bab8bd67529b9f1" }, "name" : "Celarayn", "permalink" : "celarayn",
+      "crunchbase_url" : "http://www.crunchbase.com/company/celarayn",
+      ...
+
+.. note:: Find All Documents
+
+   To find all documents in a collection, pass an empty filter 
+   to the ``find()`` method:
+
+   .. code-block:: cpp
+
+      auto cursor = collection.find({})
+
+Modify Find Behavior
+~~~~~~~~~~~~~~~~~~~~
+
+You can modify the behavior of the ``find()`` and ``find_one()`` methods by passing
+an instance of the ``mongocxx::options::find`` class as a parameter. The following table
+describes some of the fields you can set in a ``mongocxx::options::find`` instance:
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Field
+     - Description
+
+   * - ``batch_size`` 
+     - | The number of documents to return per batch.
+       | Type: ``std::int32_t``
```

Comment:
Q: Should we bold `Type:` here? No strong opinions, but just a thought

Reply:
Yeah I like that idea!